src/site/shared/utils/SnapUtils.ts

```diff
@@ -0,0 +1,40 @@
+import {Component} from "core/public";
+import {V, Vector} from "Vector";
+
+
+const WIRE_SNAP_THRESHOLD = 0.2;
+
+/**
+ * Utility used for calculating positions of ports and wires when a component
+ * gets moved.
+ *
+ * @param oPos The origin position to snap.
+ * @param comp The component to consider snapping the connections for.
+ * @returns    The snapped position for `obj`.
+ */
+export function SnapToConnections(oPos: Vector, comp: Component): Vector {
```

Comment:
Consider a more descriptive name than `oPos`.  The doc comment also isn't very helpful in understanding what it means.